### ISSUE TITLE
Error while loading shared libraries: libwebkit2gtk-4.0.so.37

### ISSUE THREAD

**Issue by mfacton** [bambulab/BambuStudio#4232]
### Bambu Studio Version

1.9.1

### Where is the application from?

Bambu Lab github releases

### OS version

Ubuntu 24.04 LTS

### Additional system information

Worked on Ubuntu 22.04, but not 24.04, even with downgrading libwebkit2gtk from 4.1 to 4.0

### Printer

Any

### How to reproduce

sudo apt-get update
sudo apt-get upgrade
sudo apt-get install libwebkit2gtk-4.0
sudo chmod +x bambu-studio.AppImage
./bambu-studio.AppImage

### Actual results

/tmp/.mount_bambu-wQgUx9/bin/bambu-studio: error while loading shared libraries: libwebkit2gtk-4.0.so.37: cannot open shared object file: No such file or directory

### Expected results

The software should open

### Project file & Debug log uploads

Does not make log (doesn't boot)

### Checklist of files to include

- [X] Log file
- [ ] Project file

**Comment by mfacton** [comment]
I just had to make symbolic links for the libwebkit2gtk-4.1 and libjavascriptcoregtk-4.1 dependencies and it works now

**Comment by EggzBacon7** [comment]
sudo apt-get install libwebkit2gtk-4.0 

isso realmente funcionou !!!

**Comment by darbdarb** [comment]
same issue on fedora

```
./Bambu_Studio_linux_fedora-v01.10.02.76.AppImage 
/tmp/.mount_Bambu_l99QjE/bin/bambu-studio: error while loading shared libraries: libwebkit2gtk-4.0.so.37: cannot open shared object file: No such file or directory
```

**Comment by EmRodDev** [comment]
Same on Fedora. The problem is that the .AppImage uses webkit2gtk4.0. This package is no longer available on Fedora 43. Please consider updating said dependency to 4.1.

**Comment by MechEng70** [comment]
Seems like the latest fedora AppImage is still looking for the webkit2gtk4.0.

./Bambu_Studio_linux_fedora-v02.04.00.70.AppImage 
/tmp/.mount_Bambu_MOMAbj/bin/bambu-studio: error while loading shared libraries: libwebkit2gtk-4.0.so.37: cannot open shared object file: No such file or directory

Flatpak is slow and zoomed in.  Frustrating.

**Comment by 2wiceUponATime** [comment]
I tried these symlinks on Fedora 43:
```
/usr/lib64/libjavascriptcoregtk-4.0.so.18 -> /usr/lib64/libjavascriptcoregtk-4.1.so
/usr/lib64/libwebkit2gtk-4.0.so.37 -> /usr/lib64/libwebkit2gtk-4.1.so
```
And got this error:
```
libsoup-ERROR **: 11:36:42.632: libsoup2 symbols detected. Using libsoup2 and libsoup3 in the same process is not supported.
```

**Comment by oggomx** [comment]
> I tried these symlinks on Fedora 43:
> 
> ```
> /usr/lib64/libjavascriptcoregtk-4.0.so.18 -> /usr/lib64/libjavascriptcoregtk-4.1.so
> /usr/lib64/libwebkit2gtk-4.0.so.37 -> /usr/lib64/libwebkit2gtk-4.1.so
> ```
> 
> And got this error:
> 
> ```
> libsoup-ERROR **: 11:36:42.632: libsoup2 symbols detected. Using libsoup2 and libsoup3 in the same process is not supported.
> ```

Same here, anyone to help us out? jaja

**Comment by Lunakin42** [comment]
Ok so I just had this issue as well. And I realized the problem was that older libraries were still there that should not be. To fix this, in addition to doing the two links above for gtk, I had to run the following:

> sudo dnf autoremove
> sudo ln -s /usr/lib64/libsoup-3.0.so.0 /usr/lib64/libsoup-2.4.so.1

The first dnf removes the old libsoup 2 so there's no conflict, but then you still need to add a symlink from it to the new version. And that worked for me.
